portal/utils.go

```diff
@@ -56,15 +61,46 @@ func getSessionRowIDAndTenantName(ctx context.Context) (string, string, error) {
 	// Get tenant name from the DB
 	getTenantShortnameQ := `SELECT s.id, t.short_name
                            FROM m3.provisioning.sessions as s JOIN m3.provisioning.tenants as t
-                           ON (s.tenant_id = t.id) WHERE s.id=$1 AND s.status=$2`
-	tenantRow := db.QueryRow(getTenantShortnameQ, sessionID, "valid")
+                           ON (s.tenant_id = t.id) WHERE s.id=$1 AND s.status=$2 AND NOW() < s.expires_at`
+	tenantRow := db.QueryRow(getTenantShortnameQ, sessionID, sessionValid)
 
 	var (
 		tenantShortname string
 		sessionRowID    string
 	)
 	err := tenantRow.Scan(&sessionRowID, &tenantShortname)
+
+	// Check if session id is expired then change status
 	if err == sql.ErrNoRows {
+		// TODO: Abstract Update functionality
+		// Get expired session_id from the DB
+		getSessionIDQ := `SELECT s.id
+                           FROM m3.provisioning.sessions as s 
+                           WHERE s.id=$1 AND s.status=$2 AND NOW() >= s.expires_at`
+
+		tenantRow := db.QueryRow(getSessionIDQ, sessionID, sessionValid)
+		err = tenantRow.Scan(&sessionRowID)
+		// If session id exists, change the status to invalid
+		if err == nil && sessionRowID != "" {
+			setSessionStatusQ :=
+				`UPDATE m3.provisioning.sessions 
```

Comment:
We shouldn't update the session status when we are fetching session related information. The session status indicates whether or not a user is logged-in while the expires_at column signifies until what time it is valid. So, there is no need for an expired session to be marked invalid.

Reply:
makes sense, let's keep both columns separate and not linked.